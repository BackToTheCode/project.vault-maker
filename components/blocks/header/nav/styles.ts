const styles = {
  container: {
    ml: 7
  },
  item: {
    '&:hover': {
      color: 'rgb(255,255,255)'
    },
    color: 'rgba(255,255,255, 0.5)',
    cursor: 'pointer',
    letterSpacing: '0.2px',
    mr: 5,
    p: 4
  },
  itemActive: {
    color: 'rgb(255,255,255)'
  }
};

export default styles;
