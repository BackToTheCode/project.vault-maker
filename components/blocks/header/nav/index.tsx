/** @jsx jsx */
import { Button } from '@backtothecode/vm-ui-library';
import { jsx } from '@emotion/core';
import { useRouter } from 'next/router';
import { FC } from 'react';
import { Flex } from 'rebass';
import styles from './styles';

export const Nav: FC<{}> = () => {
    const { container, item, itemActive} = styles;
    const router = useRouter();
    const routeToVaults = (evt: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
        evt.preventDefault();
        router.push('/vaults');
    }

    const routeToMake = (evt: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
        evt.preventDefault();
        router.push('/');
    }

  return (
    <Flex sx={container}>
        <Button onClick={routeToMake} variant={'text'} sx={{...item, ...itemActive}}>Make</Button>
        <Button onClick={routeToVaults} variant={'text'} sx={item}>Vaults</Button>
    </Flex>
  );
};

