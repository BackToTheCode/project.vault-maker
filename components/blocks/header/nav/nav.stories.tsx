import { checkA11y } from '@storybook/addon-a11y';
import { boolean, withKnobs } from '@storybook/addon-knobs';
import React from 'react';
import { Box } from 'rebass';
import { Nav } from '../nav';

const Background = ({ children }) => (
  <Box sx={{ bg: 'rgb(55,74,97)', ml: 7}}>{children}</Box>
);

export default {
  component: Nav,
  decorators: [checkA11y],
  parameters: {
    backgrounds: [
      { name: 'midGrey', value: 'rgb(55,74,97)', default: true },
      { name: 'primary', value: 'rgb(26, 171, 155)' }
    ]
  },
  title: 'Elements|Nav'
};

export const Default = () => (
  <Background>
    <Nav />
  </Background>
);