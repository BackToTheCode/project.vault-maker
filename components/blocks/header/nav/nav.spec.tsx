import { render } from '@testing-library/react';
import React from 'react';
import { Nav } from '../nav';


describe('<Nav /> spec', () => {
  test('it renders the component', () => {
    // Arrange
    const expectedText = 'Vaults';

    // Act
    const { getByText } = render(<Nav />);

    // Assert
    expect(getByText(expectedText)).toMatchSnapshot();
  });
});
