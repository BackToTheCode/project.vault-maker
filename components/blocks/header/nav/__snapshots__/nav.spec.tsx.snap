// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> spec it renders the component 1`] = `
<button
  class="css-nj2cbt"
>
  Vaults
</button>
`;
